projectroles/app_settings.py

```diff
@@ -426,6 +427,15 @@ def _log_debug(action, app_name, setting_name, value, project, user):
             if not app_name == 'projectroles':
                 q_kwargs['app_plugin__name'] = app_name
             setting = AppSetting.objects.get(**q_kwargs)
+
+            # Check if setting is set to a category
+            if (
+                setting
+                and setting.project
+                and setting.project.type == PROJECT_TYPE_CATEGORY
+            ):
+                return False
```

Comment:
Instead of returning False, I suggest we instead raise a `ValueError` with a message about settings not being allowed for categories. That makes it harder to ignore if a user attempts to misuse the feature :)

This check and potential exception raising should be in the beginning of `set()` along with the check for project and user args. This also means we only need to check this once and can remove the latter check under ` except AppSetting.DoesNotExist:`.